src/pages/Main/MainTopBanner/Maintopbanner.js

```diff
@@ -0,0 +1,143 @@
+
+  const movePrevBannerImage = e => {
+    if (currentBanner === 1) {
+      setcurrentBanner(5);
+    } else {
+      setcurrentBanner(currentBanner - 1);
+    }
+  };
+
+  const goToBannerImage = imageId => {
+    setcurrentBanner(imageId);
+  };
+
+  return (
+    <MainTopBannerStyle>
+      <div>
+        {topBannerImageIds.map((imageId, index) => {
+          return (
+            <BannerImageContainer
+              backgroundIndex={[imageId, index, currentBanner]}
+            >
+              <BannerImage
+                key={index}
+                alt="banner image"
+                imageId={[imageId, currentBanner]}
+                src={`/images/topBanner_${imageId}.jpeg`}
+              />
+            </BannerImageContainer>
+          );
+        })}
+      </div>
+      <TargetButtonList>
+        {topBannerImageIds.map((imageId, index) => {
+          return (
+            <TargetButton
+              key={index}
+              imageId={[imageId, currentBanner]}
+              onClick={() => goToBannerImage(imageId)}
+            />
+          );
+        })}
+      </TargetButtonList>
+      <PrevButton onClick={movePrevBannerImage}>
+        <i class="fas fa-chevron-left"></i>
+      </PrevButton>
+      <NextButton onClick={moveNextBannerImage}>
+        <i class="fas fa-chevron-right"></i>
+      </NextButton>
+    </MainTopBannerStyle>
+  );
+};
+
+export default MainTopBanner;
+
+const bannerBackgroundColors = [
+  '#3398dc',
+  '#255762',
+  '#263f2c',
+  '#4ebd95',
+  '#e8ddd7',
+];
+
+const MainTopBannerStyle = styled.div`
+  position: relative;
+  margin-top: 90px;
+  width: 100%;
+  height: 480px;
+`;
+
+const BannerImageContainer = styled.div`
+  ${props => props.theme.flexbox('column')};
+  position: absolute;
+  width: 100%;
+  height: 100%;
+  background-color: ${props =>
+    bannerBackgroundColors[props.backgroundIndex[1]]};
+  opacity: ${props =>
+    props.backgroundIndex[0] === props.backgroundIndex[2] ? 1 : 0};
+`;
+
+const BannerImage = styled.img`
+  position: absolute;
+  height: 480px;
+  transition: opacity 1000ms;
+  opacity: ${props => (props.imageId[0] === props.imageId[1] ? 1 : 0)};
+`;
+
+const TargetButtonList = styled.ul`
+  position: absolute;
+  bottom: 16px;
+  left: 0;
+  width: 100%;
+  text-align: center;
+`;
+const TargetButton = styled.li`
+  display: inline-block;
+  width: 8px;
+  height: 8px;
+  margin-left: 12px;
+  border-radius: 50%;
+  background-color: ${props =>
+    props.imageId[0] === props.imageId[1]
+      ? 'white'
+      : 'rgba(255, 255, 255, 0.3)'};
+`;
+
+const PrevButton = styled.button`
+  ${props => props.theme.flexbox('column')};
+  position: absolute;
+  top: 0;
+  left: 24px;
+  height: 100%;
+  .fas {
+    color: white;
+    font-size: 48px;
+  }
+`;
+
+const NextButton = styled.button`
+  ${props => props.theme.flexbox('column')};
+  position: absolute;
+  top: 0;
+  right: 24px;
+  height: 100%;
+  .fas {
+    color: white;
+    font-size: 48px;
+  }
+`;
```

Comment:
해당 부분 left와 right만 다르고 다른 스타일은 모두 같네요!
하나로 한 번 합쳐볼까요??
다른 부분만 props를 통해서 해결하면 좀 더 간결하게 표현이 가능하겠네요!!

Reply:
```suggestion
const buttonStyle = styled.button`
  ${props => {
    return props.theme.flexbox('column'), 
    props.location;
  }};
  position: absolute;
  top: 0;
  height: 100%;
  .fas {
    color: white;
    font-size: 48px;
  }
`;

const PrevButton = buttonStyle;
const NextButton = buttonStyle;
```

위 처럼 합치고, 다른 부분은 location 이란 props 명으로 각각 {left : 24px}, {right : 24px} 받아오게했는데 이렇게 하는게 맞는걸까요?